hack/README.md

```diff
@@ -0,0 +1,215 @@
+# OpenShift Shell Style Guide
+
+## Table of Contents
+1. [Introduction](#introduction)
+  1. [Which Shell To Use](#which-shell-to-use)
+  2. [When to use Shell](#when-to-use-shell)
+2. [Output](#output)
+  1. [Standard Out and Standard Error](#standard-output-and-standard-error)
+3. [Comments](#comments)
+  1. [Header](#header)
+  2. [Function Comments]($function-comments)
+  3. [Implementation Comments](#implementation-comments)
+4. [Formatting](#formatting)
+  1. [Indentation](#indentation)
+  2. [Line Length](#line-length)
+  3. [Pipelines](#pipelines)
+  4. [Loops](#loops)
+  5. [Variable Expansion](#variable-expansion)
+  6. [Quoting](#quoting)
+5. [Other Guidelines](#other-guidelines)
+  1. [Shell Script Safety](#shell-script-safety)
+  2. [Passing Arguments](#passing-arguments)
+  3. [Command Substitution](#command-substitution)
+  4. [Tests](#tests)
+  5. [Testing Strings](#testing-strings)
+  6. [Eval](#eval)
+  7. [Pipes to While](#pipes-to-while)
+6. [Naming Conventions](#naming-conventions)
+  1. [Functions](#functions)
+  2. [Variable Names](#variable-names)
+  3. [Readonly Variables](#readonly-variables)
+  4. [Local Variables](#local-variables)
+
+## Introduction
+
+### Which Shell To Use
+Bash should be the only shell used for OpenShift scripts. 
```

Comment:
Maybe call out the specific version? It needs to be Bash 3.2-compatible shell, right?